src/couch_expiring_cache/src/couch_expiring_cache.erl

```diff
@@ -0,0 +1,53 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_expiring_cache).
+
+-export([
+    insert/5,
+    insert/6,
+    lookup/2,
+    lookup/3
+]).
+
+
+-include_lib("couch_expiring_cache/include/couch_expiring_cache.hrl").
+
+
+-spec insert(Name :: binary(), Key :: binary(), Value :: binary(),
+    StaleTS :: millisecond(), ExpiresTS :: millisecond()) -> ok.
+insert(Name, Key, Value, StaleTS, ExpiresTS) ->
+    insert(undefined, Name, Key, Value, StaleTS, ExpiresTS).
+
+
+-spec insert(Tx :: jtx(), Name :: binary(), Key :: binary(), Value :: binary(),
+    StaleTS :: millisecond(), ExpiresTS :: millisecond()) -> ok.
+insert(Tx, Name, Key, Value, StaleTS, ExpiresTS) ->
+    couch_jobs_fdb:tx(couch_jobs_fdb:get_jtx(Tx), fun(JTx) ->
```

Comment:
If you want, maybe add some guards on the types of arguments. 

`is_binary/1` for `Name`, `Key` and `Value` and `is_integer/1` for `StaleTs` and `ExpiresTS`

Reply:
Done.